### PR TITLE
Dev firebase

### PR DESCRIPTION
**Pull request by HanaFarid**

### CODE REVIEW

File: app/src/main/java/com/github/leuludyha/ibdb/FirebaseActivity.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import com.github.leuludyha.ibdb.ui.theme.IBDBTheme
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import java.util.concurrent.CompletableFuture
+
+private val db: DatabaseReference = Firebase.database.reference
```
Comment (WollfieGitHub) [inline app/src/main/java/com/github/leuludyha/ibdb/FirebaseActivity.kt:22]: As it's for the bootcamp I don't think there is any problem putting it in this file only but was there any suggestion in the documentation/bootcamp instructions as how to access such an object we need in the whole application ?

File: app/src/main/java/com/github/leuludyha/ibdb/FirebaseActivity.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            val future = CompletableFuture<String>()
+
+            db.child(phone).get().addOnSuccessListener {
+                if (it.value == null) future.completeExceptionally(NoSuchFieldException())
+                else future.complete(it.value as String)
+            }.addOnFailureListener {
+                future.completeExceptionally(it)
+            }
+
+            future.thenAccept {
```
Comment (WollfieGitHub) [inline app/src/main/java/com/github/leuludyha/ibdb/FirebaseActivity.kt:78]: Maybe add exception handling if we implement this in the main code here, like painting the Button red for 1 second

File: app/src/main/java/com/github/leuludyha/ibdb/FirebaseActivity.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        Text(text = "Get")
+    }
+}
+
+@Composable
+fun SetButton(email: String, phone: String) {
+    Button(
+        onClick = {
+            if (phone.isNotEmpty()) {
+                db.child(phone).setValue(email)
```
Comment (WollfieGitHub) [inline app/src/main/java/com/github/leuludyha/ibdb/FirebaseActivity.kt:92]: Same as previous comment, if there is any connection problem or IDK

**Review by WollfieGitHub — APPROVED** [review]

**Comment by Lynoxe** [comment]
Can we close this for now and keep the branch alive since the CI doesn't pass?

File: app/src/main/java/com/github/leuludyha/ibdb/FirebaseActivity.kt (hunk 1 above)
Comment (HanaFarid) [reply app/src/main/java/com/github/leuludyha/ibdb/FirebaseActivity.kt:22]: Good point, we can keep it here for now and then move it to another file later to be able to use it in other parts. There was nothing about this in the instructions

File: app/build.gradle [hunk 4]
```diff
@@ -58,6 +59,9 @@ dependencies {
     implementation "androidx.compose.ui:ui:$compose_version"
     implementation "androidx.compose.ui:ui-tooling-preview:$compose_version"
     implementation "androidx.compose.material3:$material_version"
+    implementation 'androidx.appcompat:appcompat:1.6.1'
+    implementation 'com.google.android.material:material:1.4.0'
```
[the remark this answers is not in the document]
Comment (HanaFarid) [reply app/build.gradle]: I think you're right, I removed them

**Pull request closed without merging**